src/services/filterManager.ts

```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
   )
 }
 
+export const removeHistoryRawFromLocalStorage = (id, index, collection) => {
+  let filters = loadHistoyToLocalStorage(index, collection)
+  let idIndex = filters
+    .map(filter => {
+      return filter.id
+    })
+    .indexOf(id)
+  filters.splice(idIndex, 1)
+  localStorage.setItem(
+    `${HISTORY_LOCALSTORAGE_PREFIX}:${index}/${collection}`,
+    JSON.stringify(filters)
+  )
+}
+
+export const modifieHistoryRawFromLocalStorage = (
+  id,
+  newName,
+  bookmark,
+  index,
+  collection
+) => {
+  let filters = loadHistoyToLocalStorage(index, collection)
+  let filtersModified = filters.map(filter => {
+    if (filter.id === id) {
+      filter.name = newName
+    }
+    if (filter.id === id) {
+      filter.bookmark = bookmark
+    }
+    return filter
+  })
+  localStorage.setItem(
+    `${HISTORY_LOCALSTORAGE_PREFIX}:${index}/${collection}`,
+    JSON.stringify(filtersModified)
+  )
+}
+
+export const saveHistoyToLocalStorage = (filter, index, collection) => {
```

Comment:
Same as above: the naming suggests that the whole history is saved to localStorage (which, to me, seems the good thing to do), but if you just want to save one filter, you should call it 
```suggestion
export const saveFilterToLocalStorage = (filter, index, collection) => {
```